ONSdigital/eq-questionnaire-runner
# Update Webdriver for nanoid and fix other security issues

### What is the context of this PR?
Webdriver needed to be updated to v7 to fix a security issue a couple of layers down the dependancy tree.

- nanoid needed updating to v3.1.31 or later for the security issue
- nanoid is a dependancy of mocha
- mocha is a dependancy of mocha-framework which is part of webdriver

This PR will update Webdriver to the latest version (7.16.16) and its dependancies will now use nanoid v3.2.0.

This PR also fixes a number of other vulnerabilities highlighted by running `yarn audit`. Because of this these packages have also been updated:
- uuid
- node-jose
- eslint

For compatibility babel also needed to be updated to v7 and some changes needed to be made to the config because of this.

This also updates the node version in the github action to be the same as the one in the .nvmrc file

I have also upgraded all packages highlighted by running `yarn outdated` bar `eslint-config-standard` and `eslint-plugin-standard` as there are some compatibility issues around these with v8 of eslint, waiting on v17 of eslint-config-standard. Once that is released and we update to that we can remove `eslint-plugin-standard` as its no longer needed.

### How to review 
- Run `yarn audit` on master
- Run `yarn audit` on this branch and see all vulnerabilities are fixed
- Run `yarn outdated` on master
- Run `yarn outdated` on this branch and see nearly all packages are updated
- Functional tests pass

### Checklist
